libs/partners/openai/langchain_openai/chat_models/base.py

```diff
@@ -405,6 +406,22 @@ def _handle_openai_bad_request(e: openai.BadRequestError) -> None:
         raise
 
 
+def _is_builtin_tool(tool: dict) -> bool:
+    return set(tool.keys()) == {"type"}
+
+
+def _transform_payload_for_responses(payload: dict) -> dict:
+    updated_payload = payload.copy()
+    if messages := updated_payload.pop("messages"):
+        last_user_message = next(
+            (m for m in reversed(messages) if m.get("role") == "user"), None
+        )
+        if last_user_message:
+            updated_payload["input"] = last_user_message["content"]
```

Comment:
input can still be a list of messages no?

![Screenshot 2025-03-11 at 12 16 34 PM](https://github.com/user-attachments/assets/8b30edd6-9004-4c91-ac5b-955072370187)

Reply:
also should probably check the content block types of all messages. eg they updated type: text -> type: input_text and type: image_url -> type: input_image